app/scripts/modules/netflix/fastProperties/domain/canary.domain.ts

```diff
@@ -0,0 +1,87 @@
+
+import {PropertyCommand} from './propertyCommand.model';
+import {IUser} from 'core/authentication/authentication.service';
+import {PropertyCommandType} from './propertyCommandType.enum';
+import {ICanaryDeployment} from './canaryDeployment.interface';
+import {AcaTaskStageConfigDetails} from './acaTaskStageConfigDetails.model';
+
+export class Canary {
+  public application: string = 'spinnakerfp';
+  public owner: string;
+  public canaryConfig: CanaryConfig;
+  public watchers: string[] = [];
+  public canaryDeployments: CanaryDeployments[];
+
+  constructor(private user: IUser, private command: PropertyCommand, private configDetails: AcaTaskStageConfigDetails ) {
+    this.owner = user.name;
+    this.canaryConfig = new CanaryConfig(command);
+    this.watchers = configDetails.watchers.split(',');
+    this.canaryDeployments = [new CanaryDeployments(command.strategy.configDetails)];
+  }
+}
+
+
+class CanaryConfig {
+  public name: string;
+  public lifetimeHours: string;
+  public combinedCanaryResultStrategy: string;
+  public canaryHealthCheckHandler: CanaryHealthCheckHandler;
+  public canaryAnalysisConfig: CanaryAnalysisConfig;
+  public canarySuccessCriteria: CanarySuccessCriteria;
+
+  constructor(command: PropertyCommand) {
+    this.name = command.strategy.configDetails.name || `FP - ${PropertyCommandType[command.type]} ACA Task`;
+    command.strategy.configDetails.name = this.name;
+    this.lifetimeHours = command.strategy.configDetails.lifetimeHours;
+    this.combinedCanaryResultStrategy = command.strategy.configDetails.combinedCanaryResultStrategy;
+    this.canaryHealthCheckHandler = new CanaryHealthCheckHandler(command.strategy.configDetails.minimumCanaryResultScore);
+    this.canaryAnalysisConfig = new CanaryAnalysisConfig(command.strategy.configDetails);
+    this.canarySuccessCriteria = new CanarySuccessCriteria(command.strategy.configDetails.canaryResultScore);
+  }
+}
+
+class CanaryAnalysisConfig {
+  public notificationHours: number[] = [];
+  public useLookback: boolean;
+  public lookbackMins: number;
+  public  name: string;
+  public  beginCanaryAnalysisAfterMins: string;
+  public canaryAnalysisIntervalMins: string;
+
+  constructor(private config: any) {
```

Comment:
ditto re ctor access modifier.  please check the rest of the ctors in TS classes to confirm use.  thanks.